include/revng/StackAnalysis/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1089 @@
+  ///
+  /// If the argument that doen't fit into the allowed register count it is
+  /// placed on the stack.
+  ///
+  /// \note: some special cases still apply. For example C++ abi doesn't allow
+  /// structs and classes that are not 'trivial for the purposes of calls' to
+  /// be passed in such a fashion. As such, object with irregular padding,
+  /// non-trivial construction/destruction, etc are only passed using the stack.
+  /// (see https://itanium-cxx-abi.github.io/cxx-abi)
+  static constexpr size_t MaxGeneralPurposeRegistersPerAggregateArgument = 0;
+
+  /// States whether specified ABI allows returning aggregate objects (structs
+  /// unions, etc) in the designated registers if they provide enough space
+  /// space (including padding) and the maximum number of registers allowed for
+  /// those values.
+  ///
+  /// The rules from `MaxGeneralPurposeRegistersPerAggregateArgument` apply,
+  /// see relevant comment for additional information.
+  /// \sa MaxGeneralPurposeRegistersPerAggregateArgument
+  static constexpr size_t MaxGeneralPurposeRegistersPerAggregateReturnValue = 0;
+
+  /// States whether specified ABI allows accepting primite object arguments
+  /// (`int`s, `long`s, `__int128`s, etc) in the designated registers if they
+  /// provide enough space (including padding) and the maximum number of
+  /// registers allowed per argument.
+  ///
+  /// `0` means that using general purpose registers is not allowed.
+  ///
+  /// `1` means that passing values is only allowed if their size doesn't
+  /// exceed the size of a single general purpose register.
+  ///
+  /// `2` means that passing aggregate values is only allowed if their size
+  /// doesn't exceed the size of two general purpose registers.
+  ///
+  /// ... and so on.
+  ///
+  /// if `MaxGeneralPurposeRegistersPerPrimitiveArgument` is greater than or
+  /// equal to `GeneralPurposeArgumentRegisters.size()`, all the general purpose
+  /// registers allowed to be used for function arguments could be used for
+  /// a single aggregate argument.
+  ///
+  /// If the argument that doesn't fit into the allowed register count it is
+  /// placed on the stack.
+  static constexpr size_t MaxGeneralPurposeRegistersPerPrimitiveArgument = 0;
+
+  /// States whether specified ABI allows returning primitive objects (`int`s,
+  /// `long`s, `__int128`s, etc) in the designated registers if they provide
+  /// enough space and the maximum number of registers allowed per argument.
+  ///
+  /// The rules from `MaxGeneralPurposeRegistersPerPrimitiveArgument` apply,
+  /// see relevant comment for additional information.
+  /// \sa MaxGeneralPurposeRegistersPerPrimitiveArgument
+  static constexpr size_t MaxGeneralPurposeRegistersPerPrimitiveReturnValue = 0;
+
+  /// States who is responsible for cleaning the stack after the function call.
+  /// `true` - callee, `false` - caller.
+  static constexpr bool CalleeIsResponsibleForStackCleanup = false;
```

Comment:
This will affect the `FinalStackOffset`. Are you already taking this into account?

Reply:
No, as of now, the options is unused since stack support is practically non-existent at this point. The option itself is kept so that searching through all the documents wouldn't be necessary when we do actually find something depending on the stack cleanup.